wasmplugin/plugin.go

```diff
@@ -223,6 +235,22 @@ func (ctx *httpContext) OnHttpRequestBody(bodySize int, endOfStream bool) types.
 
 func (ctx *httpContext) OnHttpResponseHeaders(numHeaders int, endOfStream bool) types.Action {
 	defer logTime("OnHttpResponseHeaders", currentTime())
+
+	if ctx.interruptionHandled {
+		// Handling the interruption (see handleInterruption) generates a HttpResponse with the required status code.
+		// If handleInterruption is raised during OnHttpRequestHeaders or OnHttpRequestBody, the crafted response is sent
+		// downstream via the filter chain, therefore OnHttpResponseHeaders is called.
+		// We expect a response that is ending the stream, with exactly one header (:status) and no body.
+		// See https://github.com/corazawaf/coraza-proxy-wasm/pull/126
+		if numHeaders == 1 && endOfStream {
```

Comment:
I went for the first approach allowing only OnHttpResponseHeaders to return an `ActionContinue` after that the interruption has been handled. The returned response received should be the local response just generated by handleInterruption.
If we want to be more conservative, we may turn `interruptionHandled` into an `int` and check that the single header (:status) is populated with the expected interruption code